packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -21,6 +21,9 @@ import 'widget_inspector.dart';
 
 export 'dart:ui' show AppLifecycleState, Locale;
 
+/// A callback that can be registered with [WidgetsBinding.registerHotRestartCallback].
+typedef PreHotRestartCallback = Object? Function();
```

Comment:
This should document what the return value should be. It's very unusual for an API to just allow any return value.

(I don't understand why this can't just be `void` or `Future<void>` though?)